vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -40,60 +32,33 @@
 
 logger = init_logger(__name__)
 
-TModelInputForSpyre = TypeVar('TModelInputForSpyre',
-                              bound="ModelInputForSpyre")
-
-
-@dataclass(frozen=True)
-class ModelInputForSpyre(ModelRunnerInputBase):
```

Comment:
Should we keep this as a slimmer dataclass with just the things we need in it? That way we can at least keep one container as the return from `prepare_model_input` instead of returning 5 things:
```
        input_tokens, input_positions, input_masks, \
            sampling_metadata, is_prompt = (
            self.prepare_model_input(scheduler_output))
``` 

We can nix the unused `virtual_engine` and the `pooling_metadata` since this won't run embeddings, along with the broadcasting functions that were used in v0

Reply:
Yeah I thought about that, the only reason I went against that decision was because of naming conventions - calling it something like `ModelInput` might confuse folks. Perhaps a unique disambiguous name would be helpful

Reply:
Gave it a shot, let me know if it looks better!